### PR TITLE
Treat near-zero intercept values as zero when calculating weights

### PR DESCRIPTION
**Pull request by koute**
Followup of https://github.com/paritytech/substrate/pull/12482 (in particular this comment: https://github.com/paritytech/substrate/pull/12482#discussion_r1002097077)

This PR makes it so that zero or near zero intercept values generated when running the linear regression will be rounded up to zero and emitted as-is as base weights.

This is a very specific corner case. In practice this should most likely not make much of a difference. Since we're dealing with floating point numbers we'll essentially *never* have an intercept of *exactly* zero; however, it is practically possible to have a near-zero intercept which is only very slightly negative due to numerical inaccuracies (see the test I've added for an example) which I imagine could *maybe* happen for e.g. the database read/write weights or any other entirely deterministic measure.

### CODE REVIEW

**Review by shawntabrizi — APPROVED** [review]

File: frame/benchmarking/src/analysis.rs [hunk 1]
```diff
@@ -170,7 +170,8 @@ fn linear_regression(
 	x_vars: usize,
 ) -> Option<(f64, Vec<f64>, Vec<f64>)> {
 	let (intercept, params, errors) = raw_linear_regression(&xs, &ys, x_vars, true)?;
-	if intercept > 0.0 {
+	if intercept > 0.0 || intercept.abs() <= 0.0001 {
```
Comment (ggwpez) [inline frame/benchmarking/src/analysis.rs:173]:
```suggestion
	if intercept >= -0.0001 {
```
A bit shorter?

**Review by ggwpez — APPROVED** [review]

**Comment by ggwpez** [comment]
/cmd queue -c bench-bot $ pallet dev pallet_contracts

**Comment by koute** [comment]
Looks like the bot's broken; the docker image probably has to be updated with newer version of Rust?

**Comment by koute** [comment]
bot merge

**Pull request merged**